### DISCUSSION TITLE
2 copies of one cell (sames parameters), but only one cell name.

### DISCUSSION THREAD

**Discussion by tmerelle** [gdsfactory/gdsfactory#2913] [Q&A]
Hi !

I created a GDS file that calls twice the same cell, using the same parameters.
And though the called function used to generate these two cell copies has a @cell decorator, the cell appears only once in the list of cell names in klayout (see yellow arrow), while as expected it does show up twice at the right locations on the layout itself, with exactly the same name (see the 2 green arrows).
The thing is: I really need to get different names for those 2 cells for GDS file import use in another environment. 

See the picture below:
![Example](https://github.com/gdsfactory/gdsfactory/assets/64286076/2af4246c-bc1f-4505-9b24-70fc9fe84562)

Thomas

**Comment by joamatab** [comment]
Hi Thomas, this is normal,

if the cell is the same, the two instances point in memory to the same cell

**Comment by sebastian-goeldi** [comment]
Hi Thomas,

As Joaquin already said, a cell/component can (and often will) have multiple instances/references.

You can force the creation of new cells prior to making instances, but __not__ with the same exact name. There can only be one cell per unique name as its the unique identifier of the cell.

You could enforce that with

```python
inst1 = c << my_cell.dup() # creates cell with name "{my_cell.name}$1 and makes an instance of it"
inst2 = c << my_cell.dup() # creates cell with name "{my_cell.name}$2 and makes an instance of it"
```

be very cautious with this though as this will cause the creation of a new cell every time this code is called.  This can get out of hand very fast.

May I ask why you want to create new cells here? Usually that is very much not desired.
